### ISSUE TITLE
[Bug] Moving bookmarks into new folders retains their position

### ISSUE THREAD

**Issue by grigoryk** [mozilla-mobile/fenix#15211]
Bookmarks have a "position" property, which is their place within a folder relative to other bookmarks in that folder.

Currently, when a bookmark is moved that position is left as-is. Try moving a bookmark from a small folder into a large folder. Since position will be retained, bookmark will appear somewhere in the beginning or towards the middle of the new folder.

This behaviour makes it feel like bookmarks are "lost" after moves, while they're actually just hiding somewhere in the middle.

We should be changing position so that bookmark is either at the very end, or at the very beginning of the folder.

From places API docs:
```
If `position` is not provided (and `parentGUID` is) then its treated as a move to the end of that folder.`
```

So let's just do that.

**Comment by grigoryk** [comment]
Should be fixed by https://github.com/mozilla-mobile/fenix/pull/15212

**Comment by ebalazs-sv** [comment]
Verified as fixed on Nightly 200921 06:05 (Build #2015765171) GV 82.0a1 from 9/21 with Motorola Moto G6 (Android 8), Google Pixel 2 (Android 9), Nexus 5 (Android 6.0.1). Moving a bookmark into a new folder is placing it at the very end of that folder.
I will remove the qa:needed label and close this issue.
